walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,61 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<Event> existingEvents = new ArrayList<Event>(events);
```

Comment:
Although the logic handles it well, I think it can be improved further by doing it as separate validation logic at the beginning.
The current approach needs to iterate all the events unnecessarily for a meeting request that has no attendee or the duration is too long.

Reply:
Added additional validation logic for edge cases